### PR TITLE
Full translation of chapter 6 of the course.

### PR DESCRIPTION
**Pull request by blademoon**
#75

@MKhalusova

Good day Maria. Full translation of chapter 6 of the course. If you have any comments, tag me. I look forward to your review.

### CODE REVIEW

File: chapters/ru/chapter6/1.mdx
Comment (blademoon) [inline chapters/ru/chapter6/1.mdx:1]: I just re-translated the whole thing. To keep the translation consistent throughout the chapter.

File: chapters/ru/chapter6/2.mdx
Comment (blademoon) [inline chapters/ru/chapter6/2.mdx:1]: I just re-translated the whole thing. To keep the translation consistent throughout the chapter.

File: chapters/ru/chapter6/2.mdx [hunk 1]
```diff
@@ -217,41 +217,41 @@ tokenizer.tokenize(example)
...
 
-После входа в систему вы можете активировать свой токенизатор, выполнив следующую команду:
+После того как вы авторизовались, вы можете опубликовать свой токенизатор, выполнив следующую команду:
 
 ```py
 tokenizer.push_to_hub("code-search-net-tokenizer")
 ```
 
-Это создаст новый репозиторий в вашем пространстве имен с именем `code-search-net-tokenizer`, содержащий файл токенизатора. Затем вы можете загрузить токенизатор из любого места с помощью метода  `from_pretrained()`:
+Это создаст новое хранилище в вашем пространстве имен с именем `code-search-net-tokenizer`, содержащее файл токенизатора. Затем вы можете загрузить токенизатор где угодно с помощью метода `from_pretrained()`:
```
Comment (MKhalusova) [inline chapters/ru/chapter6/2.mdx:250]: Is it common to call a repository "хранилище"? I find the "репозиторий" is more common.

File: chapters/ru/chapter6/3.mdx [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+
+```py
+encoding.word_ids()
+```
+
+```python out
+[None, 0, 1, 2, 3, 3, 3, 3, 4, 5, 6, 7, 8, 8, 9, 10, 11, 12, None]
+```
+
+We can see that the tokenizer's special tokens `[CLS]` and `[SEP]` are mapped to `None`, and then each token is mapped to the word it originates from. This is especially useful to determine if a token is at the start of a word or if two tokens are in the same word. We could rely on the `##` prefix for that, but it only works for BERT-like tokenizers; this method works for any type of tokenizer as long as it's a fast one. In the next chapter, we'll see how we can use this capability to apply the labels we have for each word properly to the tokens in tasks like named entity recognition (NER) and part-of-speech (POS) tagging. We can also use it to mask all the tokens coming from the same word in masked language modeling (a technique called _whole word masking_).
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:108]:
```suggestion
```

File: chapters/ru/chapter6/3.mdx [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+
+<Tip>
+
+Понятие "слово" очень сложное. Например, "I'll" (сокращение от "I will") считается одним или двумя словами? На самом деле это зависит от токенизатора и применяемой им операции предварительной токенизации. Некоторые токенизаторы просто разделяют пробелы, поэтому они будут считать это одним словом. Другие используют пунктуацию поверх пробелов, поэтому будут считать это двумя словами.
+
+✏️ **Попробуйте! ** Создайте токенизатор из контрольных точек `bert-base-cased` и `roberta-base` и токенизируйте с их помощью "81s". Что вы заметили? Каковы идентификаторы слов?
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:116]:
```suggestion
✏️ **Попробуйте!** Создайте токенизатор из контрольных точек `bert-base-cased` и `roberta-base` и токенизируйте с их помощью "81s". Что вы заметили? Каковы идентификаторы слов?
```

File: chapters/ru/chapter6/3.mdx [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+
+<Tip>
+
+✏️ **Попробуйте!** Создайте свой собственный пример текста и посмотрите, сможете ли вы понять, какие токены связаны с идентификаторами слов, а также как извлечь диапазоны символов для одного слова. Чтобы получить бонусные очки, попробуйте использовать два предложения в качестве входных данных и посмотрите, будут ли идентификаторы предложений иметь для вас смысл.
+
+</Tip>
+
+## Внутри конвейера `token-classification`[[inside-the-token-classification-pipeline]]
+
+В [Главе 1](/course/chapter1) мы впервые попробовали применить NER - когда задача состоит в том, чтобы определить, какие части текста соответствуют сущностям, таким как люди, места или организации - с помощью функции 🤗 Transformers `pipeline()`. Затем, в [Главе 2] (/course/chapter2), мы увидели, как конвейер объединяет три этапа, необходимые для получения прогнозов из необработанного текста: токенизацию, прохождение входных данных через модель и постобработку. Первые два шага в конвейере `token-classification` такие же, как и в любом другом конвейере, но постобработка немного сложнее - давайте посмотрим, как это сделать!
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:143]:
```suggestion
В [Главе 1](/course/chapter1) мы впервые попробовали применить NER - когда задача состоит в том, чтобы определить, какие части текста соответствуют сущностям, таким как люди, места или организации - с помощью функции 🤗 Transformers `pipeline()`. Затем, в [Главе 2](/course/chapter2), мы увидели, как конвейер объединяет три этапа, необходимые для получения прогнозов из необработанного текста: токенизацию, прохождение входных данных через модель и постобработку. Первые два шага в конвейере `token-classification` такие же, как и в любом другом конвейере, но постобработка немного сложнее - давайте посмотрим, как это сделать!
```

File: chapters/ru/chapter6/3.mdx [hunk 5]
```diff
@@ -0,0 +1,475 @@
...
+ {'entity_group': 'LOC', 'score': 0.99321055, 'word': 'Brooklyn', 'start': 49, 'end': 57}]
+```
+
+Выбранная `aggregation_strategy` изменит оценки, вычисляемые для каждой сгруппированной сущности. При использовании значения `"simple"` оценка является средним значением оценок каждого токена данной сущности: например, оценка "Sylvain" является средним значением оценок, которые мы видели в предыдущем примере для токенов `S`, `##yl`, `##va` и `##in`. Другие доступные стратегии:
+
+- `"first"`, где оценка каждой сущности - это оценка первого токена этой сущности (так, для "Sylvain" это будет 0,993828, оценки токена `S`)
+- `"max"`, где оценка каждой сущности - это максимальная оценка токенов в этой сущности (так, для ""Hugging Face"" это будет 0.98879766, оценки "Face").
+- `"average"`, где оценка каждой сущности - это средняя оценка слов, составляющих эту сущность (таким образом, для слова ""Sylvain"" не будет никаких отличий от стратегии `"simple"`, но "Hugging Face" будет иметь оценку 0.9819, среднюю оценку для "Hugging", 0.975, и "Face", 0.98879)
+
+Now let's see how to obtain these results without using the `pipeline()` function!
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:198]: This sentence needs to be translated.

File: chapters/ru/chapter6/3b.mdx [hunk 6]
```diff
@@ -0,0 +1,642 @@
...
+
+```python out
+[(0, 18, 0.33867), (173, 184, 0.97149)]
+```
+
+Эти два кандидата соответствуют лучшим ответам, которые модель смогла найти в каждом фрагменте. Модель гораздо больше уверена в том, что правильный ответ находится во второй части (это хороший знак!). Теперь нам нужно сопоставить эти два диапазона токенов с диапазонами символов в контексте (для получения ответа нам нужно сопоставить только второй, но интересно посмотреть, что модель выбрала в первом фрагменте).
+
+<Tip>
+
+✏️ **Попробуйте! ** Адаптируйте приведенный выше код, чтобы он возвращал оценки и промежутки для пяти наиболее вероятных ответов (в целом, а не по частям).
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3b.mdx:613]:
```suggestion
✏️ **Попробуйте!** Адаптируйте приведенный выше код, чтобы он возвращал оценки и промежутки для пяти наиболее вероятных ответов (в целом, а не по частям).
```

File: chapters/ru/chapter6/3b.mdx [hunk 7]
```diff
@@ -0,0 +1,642 @@
...
+```python out
+{'answer': '\n🤗 Transformers: State of the Art NLP', 'start': 0, 'end': 37, 'score': 0.33867}
+{'answer': 'Jax, PyTorch and TensorFlow', 'start': 1892, 'end': 1919, 'score': 0.97149}
+```
+
+Если мы проигнорируем первый результат, то получим тот же результат, что и в нашем конвейере для этого длинного контекста - ура!
+
+<Tip>
+
+✏️ **Попробуйте! ** Используйте лучшие оценки, которые вы вычислили ранее, чтобы показать пять наиболее вероятных ответов (для всего контекста, а не для каждого фрагмента). Чтобы проверить результаты, вернитесь к первому конвейеру и передайте `top_k=5` при его вызове.
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3b.mdx:638]:
```suggestion
✏️ **Попробуйте!** Используйте лучшие оценки, которые вы вычислили ранее, чтобы показать пять наиболее вероятных ответов (для всего контекста, а не для каждого фрагмента). Чтобы проверить результаты, вернитесь к первому конвейеру и передайте `top_k=5` при его вызове.
```

File: chapters/ru/chapter6/4.mdx [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+
+```python out
+'hello how are u?'
+```
+
+В этом примере, поскольку мы выбрали контрольную точку `bert-base-uncased`, нормализация применила нижний регистр и удалила ударения. 
+
+<Tip>
+
+✏️ **Попробуйте! ** Загрузите токенизатор из контрольной точки `bert-base-cased` и передайте ему тот же пример. Какие основные различия вы можете увидеть между версией токенизатора cased и uncased?
```
Comment (MKhalusova) [inline chapters/ru/chapter6/4.mdx:52]:
```suggestion
✏️ **Попробуйте!** Загрузите токенизатор из контрольной точки `bert-base-cased` и передайте ему тот же пример. Какие основные различия вы можете увидеть между версией токенизатора cased и uncased?
```

File: chapters/ru/chapter6/4.mdx [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+
+</Tip>
+
+## Предварительная токенизация[[pre-tokenization]]
+
+<Youtube id="grlLV8AIXug"/>
+
+Как мы увидим в следующих разделах, токенизатор не может быть обучен только на сыром тексте. Сначала необходимо разбить текст на небольшие части, например, на слова. Именно в этом и заключается этап предварительной токенизации. Как мы видели в [Главе 2] (/course/chapter2), токенизатор на основе слов (word-based tokenizer) может просто разбить необработанный текст на слова по пробелам и знакам пунктуации. Эти слова станут границами подтокенов, которые токенизатор сможет выучить в процессе обучения.
```
Comment (MKhalusova) [inline chapters/ru/chapter6/4.mdx:60]:
```suggestion
Как мы увидим в следующих разделах, токенизатор не может быть обучен только на сыром тексте. Сначала необходимо разбить текст на небольшие части, например, на слова. Именно в этом и заключается этап предварительной токенизации. Как мы видели в [Главе 2](/course/chapter2), токенизатор на основе слов (word-based tokenizer) может просто разбить необработанный текст на слова по пробелам и знакам пунктуации. Эти слова станут границами подтокенов, которые токенизатор сможет выучить в процессе обучения.
```

File: chapters/ru/chapter6/5.mdx [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+("u", "g") -> "ug"
+("u", "n") -> "un"
+("h", "ug") -> "hug"
+```
+
+Слово `"bug"` будет токенизировано как `["b", "ug"]`. Слово `"mug"`, однако, будет токенизировано как `["[UNK]", "ug"]`, поскольку буква `"m"` отсутствует в базовом словаре. Аналогично, слово `"thug" будет токенизировано как `["[UNK]", "hug"]`: буква `"t" отсутствует в базовом словаре, и применение правил слияния приводит сначала к слиянию `"u"` и `"g"`, а затем к слиянию `"h"` и `"ug"`.
+
+<Tip>
+
+✏️ ** Теперь ваша очередь! ** Как вы думаете, как будет токенизировано слово `'unhug''?
```
Comment (MKhalusova) [inline chapters/ru/chapter6/5.mdx:104]:
```suggestion
✏️ ** Теперь ваша очередь!** Как вы думаете, как будет токенизировано слово `'unhug'`?
```

File: chapters/ru/chapter6/7.mdx [hunk 11]
```diff
@@ -0,0 +1,381 @@
...
+
+```python
+import copy
+
+
+def compute_scores(model):
+    scores = {}
+    model_loss = compute_loss(model)
+    for token, score in model.items():
+        # We always keep tokens of length 1
```
Comment (MKhalusova) [inline chapters/ru/chapter6/7.mdx:319]: In some places code comments are translated, in other places they are not. Perhaps, it's better to be consistent and translate all of them?

File: chapters/ru/chapter6/7.mdx [hunk 12]
```diff
@@ -0,0 +1,381 @@
...
+</Tip>
+
+Когда этот процесс завершиться, останется только добавить в словарь специальные токены, используемые моделью, а затем итерироваться, пока мы не вычеркнем из словаря достаточно токенов, чтобы достичь желаемого размера:
+
+```python
+percent_to_remove = 0.1
+while len(model) > 100:
+    scores = compute_scores(model)
+    sorted_scores = sorted(scores.items(), key=lambda x: x[1])
+    # Remove percent_to_remove tokens with the lowest scores.
```
Comment (MKhalusova) [inline chapters/ru/chapter6/7.mdx:356]: Untranslated code comment

File: chapters/ru/chapter6/3.mdx [hunk 13]
```diff
@@ -0,0 +1,475 @@
...
+inputs_with_offsets = tokenizer(example, return_offsets_mapping=True)
+tokens = inputs_with_offsets.tokens()
+offsets = inputs_with_offsets["offset_mapping"]
+
+idx = 0
+while idx < len(predictions):
+    pred = predictions[idx]
+    label = model.config.id2label[pred]
+    if label != "O":
+        # Remove the B- or I-
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:436]: Untranslated code comment.

File: chapters/ru/chapter6/3.mdx [hunk 14]
```diff
@@ -0,0 +1,475 @@
...
+        label = label[2:]
+        start, _ = offsets[idx]
+
+        # Grab all the tokens labeled with I-label
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:440]: Untranslated code comment.

File: chapters/ru/chapter6/3.mdx [hunk 15]
```diff
@@ -0,0 +1,475 @@
...
+        all_scores = []
+        while (
+            idx < len(predictions)
+            and model.config.id2label[predictions[idx]] == f"I-{label}"
+        ):
+            all_scores.append(probabilities[idx][pred])
+            _, end = offsets[idx]
+            idx += 1
+
+        # The score is the mean of all the scores of the tokens in that grouped entity
```
Comment (MKhalusova) [inline chapters/ru/chapter6/3.mdx:450]: Untranslated code comment.

**Review by MKhalusova — COMMENTED** [review]
Thanks for working on this chapter! Overall, looks good to me, I left some minor nits.

File: chapters/ru/chapter6/1.mdx [hunk 16]
```diff
@@ -1,19 +1,19 @@
-# Введение
+# Введение[[introduction]]
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter6/1.mdx:1]: This `[[introduction]]` was in the English version. 

https://github.com/huggingface/course/blob/main/chapters/en/chapter6/1.mdx

Honestly didn't quite understand the question.

File: chapters/ru/chapter6/4.mdx (hunk 8 above)
Comment (blademoon) [reply chapters/ru/chapter6/4.mdx:52]: Yes, that space is redundant here. You're right about that.

File: chapters/ru/chapter6/1.mdx [hunk 17]
```diff
@@ -1,19 +1,19 @@
...
 />
 
-В [главе 3](/course/ru/chapter3), мы рассмотрели, как настроить модель под конкретную задачу. Когда мы это делаем, мы используем тот же токенизатор, с помощью которого была предварительно обучена модель, но что нам делать, когда мы хотим обучить модель с нуля? В этих случаях использование токенизатора, предварительно обученного на корпусе из другого домена или языка, обычно неоптимально. Например, токенизатор, обученный на английском корпусе, будет плохо работать с корпусом японских текстов, поскольку использование пробелов и пунктуации в этих двух языках сильно различается.
+В [Главе 3](/course/chapter3) мы рассмотрели, как дообучить модель для конкретной задачи. При этом мы используем тот же токенизатор, на котором была предварительно обучена модель, но что делать, когда мы хотим обучить модель с нуля? В таких случаях использование токенизатора, который был предварительно обучен на корпусе из другой области или языка, как правило, является неоптимальным. Например, токенизатор, обученный на корпусе английских текстов, будет плохо работать на корпусе японских текстов, поскольку использование пробелов и знаков препинания в этих двух языках сильно отличается.
```
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter6/1.mdx:8]: What link is the problem with? I don't quite get it.

File: chapters/ru/chapter6/3.mdx (hunk 3 above)
Comment (blademoon) [reply chapters/ru/chapter6/3.mdx:116]: There's an extra space here. You're right.

File: chapters/ru/chapter6/2.mdx (hunk 1 above)
Comment (blademoon) [reply chapters/ru/chapter6/2.mdx:250]: Yes, indeed this option would be better.Let's keep "репозиторий"

File: chapters/ru/chapter6/3.mdx (hunk 13 above)
Comment (blademoon) [reply chapters/ru/chapter6/3.mdx:435]: Corrected.

File: chapters/ru/chapter6/3.mdx (hunk 14 above)
Comment (blademoon) [reply chapters/ru/chapter6/3.mdx:439]: Corrected.

File: chapters/ru/chapter6/7.mdx (hunk 11 above)
Comment (blademoon) [reply chapters/ru/chapter6/7.mdx]: Corrected. Thanks.

File: chapters/ru/chapter6/3.mdx (hunk 5 above)
Comment (blademoon) [reply chapters/ru/chapter6/3.mdx]: Done.

File: chapters/ru/chapter6/3b.mdx (hunk 7 above)
Comment (blademoon) [reply chapters/ru/chapter6/3b.mdx:638]: Yes, an extra space.

File: chapters/ru/chapter6/3.mdx (hunk 15 above)
Comment (blademoon) [reply chapters/ru/chapter6/3.mdx]: Translated what was missing.

File: chapters/ru/chapter6/1.mdx (hunk 16 above)
[the remark this answers is not in the document]
Comment (blademoon) [reply chapters/ru/chapter6/1.mdx:1]: If I understand the problem correctly, it's not a bug. It came from the original notebook in English.
And this is at the beginning of every mdx file in the original English version of MDX files.

https://github.com/huggingface/course/blob/main/chapters/en/chapter6/1.mdx
https://github.com/huggingface/course/blob/main/chapters/en/chapter6/2.mdx
https://github.com/huggingface/course/blob/main/chapters/en/chapter6/3.mdx

I take it as service information necessary to build documentation, so I don't touch it.

**Comment by blademoon** [comment]
@MKhalusova If all comments are resolved, then we can make that PR?

**Comment by MKhalusova** [comment]
Thank you for this massive work, and for your patience. It's good to merge now.

**Review by MKhalusova — APPROVED** [review]

**Pull request merged**